common/views/components/OptIn/OptIn.js

```diff
@@ -0,0 +1,150 @@
+// @flow
+import { useState, useEffect, useContext } from 'react';
+import styled from 'styled-components';
+import cookie from 'cookie-cutter';
+import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';
+import { london } from '@weco/common/utils/format-date';
+import { font, classNames } from '@weco/common/utils/classnames';
+import Space from '@weco/common/views/components/styled/Space';
+import Icon from '../Icon/Icon';
+
+const OptInContainer = styled(Space).attrs(props => ({
+  as: 'div',
+  h: { size: 'l', properties: ['margin-bottom'] },
+  className: font('hnl', 5),
+}))`
+  background: ${props => props.theme.colors.purple};
+  color: ${props => props.theme.colors.white};
+  position: relative;
+  &:before {
+    content: '';
+    position: absolute;
+    left: 0;
+    top: 100%;
+    border-bottom: 18px solid transparent;
+    border-left: 16px solid ${props => props.theme.colors.purple};
+  }
+`;
+
+const OptInControl = styled.button.attrs({
+  className: classNames({
+    'plain-button no-margin no-padding flex-inline flex--v-center ': true,
+  }),
+})`
+  text-decoration: underline;
+  color: ${props => props.theme.colors.white};
+  cursor: pointer;
+  position: ${props => (props.optOut ? 'absolute' : 'static')};
+  top: ${props => `${props.theme.spacingUnit}px`};
+  right: ${props => `${props.theme.spacingUnit}px`};
+  &:focus,
+  &:hover {
+    text-decoration: none;
+  }
+`;
+
+const HiddenText = styled.span.attrs({
+  className: 'visually-hidden',
+})``;
+
+type Props = {|
+  text: {|
+    defaultMessage: string[],
```

Comment:
Misunderstood what this meant, normally default meaning fallback.
Is this the callout?

Reply:
The message changes depending on whether your opted in or not.
From this: 
<img width="616" alt="Screenshot 2019-10-31 at 16 01 33" src="https://user-images.githubusercontent.com/6051896/67964344-0a9c7e00-fbf8-11e9-89e0-65979ddd3111.png">

To this:
<img width="614" alt="Screenshot 2019-10-31 at 16 01 50" src="https://user-images.githubusercontent.com/6051896/67964356-0ff9c880-fbf8-11e9-8e75-bb477f0027d0.png">

defaultMessage is the one you get if you're not opted in. Would you rather I call it calloutMessage?

Reply:
What about `optInMessage` and `optedInMessage`?

Reply:
Bikeshedding question around whether we need this abstracted if we're just using it in one place, and we can think about the abstraction when we have more use cases.

Reply:
I'll make it static for now, then don't have to worry about the naming